### PR TITLE
Global Pause Test Case 

### PR DESCRIPTION
**Pull request by SubhajitPalKeysight**
### Description of PR

Summary:
Fixes # 2197 ( Addition of Test Case - Global Pause into PFC Test Plan )

### Type of change

Documentation of Test Case

Test Case - Objective
Test Case - Configuration
Test Steps

### CODE REVIEW

File: docs/PFC-test-plan.md [hunk 1]
```diff
@@ -20,13 +20,17 @@
       - [Test Objective](#test-objective-2)
       - [Test Configuration](#test-configuration-2)
       - [Test Steps](#test-steps-2)
+    - [Test Case #3 - GLOBAL PAUSE (IEEE 802.3x port based Flow Control)](#test-case-3---global-pause-ieee-8023x-port-based-flow-control)
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:23]: Change all 'port-based' to 'link-level'

File: docs/PFC-test-plan.md [hunk 2]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
    * Keysight Rx port should receive all the 'Background data traffic' as well as 'Test data traffic'. There should not be any loss observed.
 5. Stop the PFC PAUSE storm.
 6. Repeat the test with a different lossless priority value (!=Pi).
+   
+### Test Case #3 - GLOBAL PAUSE (IEEE 802.3x port based Flow Control)
+
+<b>Note</b> : 802.3x Port based Flow Control is different than IEEE 802.1Qbb PFC (Priority Flow Control)
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:261]: port-based -> link-level
Priority Flow Control -> Priority-based Flow Control

File: docs/PFC-test-plan.md [hunk 3]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+#### Test Objective
+Verify Device Under Test (DUT) processes the GLOBAL PAUSE frame.
+#### Test Configuration
+
+- On SONiC Device Under Test (DUT) configure  lossless priority of all values Pi (0 \<= i \<=
+    7).
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:267]: We don't need to enable PFC on all the priorities. We just need to enable PFC on required priorities, e.g., 3 and 4.

File: docs/PFC-test-plan.md [hunk 4]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+- Configure following traffic items on the Keysight device:
+  1. Test data traffic: A traffic item from the Keysight Tx port with all the 8 priorities Pi (0<= i <= 7), DSCP values from (0 - 63) and enable flow group by PFC Queue. Traffic should be configured with 100% line rate.
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:269]: The description of test data traffic is a little bit unclear. You can change it to "A traffic item from the Keysight Tx port to Rx port. The traffic item uses all the 64 DSCP values (0-63). PFC is enabled at all the 8 priorities of Tx port. The traffic demand is 100% line rate."

File: docs/PFC-test-plan.md [hunk 5]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+  2. GLOBAL PAUSE storm: Persistent Global pause frames from the Keysight Rx port to the Keysight Tx port. And the inter-frame transmission interval should be smaller than per-frame global pause duration.
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:270]: per-frame global pause duration -> per-frame pause duration

File: docs/PFC-test-plan.md [hunk 6]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+   
+#### Test Steps
+
+1. Start GLOBAL PAUSE storm.
+2. After a fixed duration (eg. 1 sec), start the Test data traffic.
+3. Keep the Test data traffic running for a fixed duration (eg 5sec) and then stop the traffic item.
+4. Verify the following:
+   * Keysight Rx port should receive all the 'Test data traffic'. There should not be any loss observed. Throughput should be close to link capacity.
+   * SONiC DUT will discard all Global pause frames.
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:279]: There is no way to check if SONiC will discard all global pause frames.

File: docs/PFC-test-plan.md [hunk 7]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+5. Stop the GLOBAL PAUSE storm.
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:280]: Change the order of 4 and 5.

**Review by baiwei0427 — CHANGES_REQUESTED** [review]

File: docs/PFC-test-plan.md (hunk 1 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: incorporated

File: docs/PFC-test-plan.md (hunk 3 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: Taken care of

File: docs/PFC-test-plan.md (hunk 4 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: Thank. Done

File: docs/PFC-test-plan.md (hunk 5 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: Done.

File: docs/PFC-test-plan.md (hunk 7 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: Done.

**Comment by baiwei0427** [comment]
retest this please

File: docs/PFC-test-plan.md [hunk 8]
```diff
@@ -20,13 +20,17 @@
       - [Test Objective](#test-objective-2)
       - [Test Configuration](#test-configuration-2)
       - [Test Steps](#test-steps-2)
+    - [Test Case #3 - GLOBAL PAUSE (IEEE 802.3x link level Flow Control)](#test-case-3---global-pause-ieee-8023x-link-level-flow-control)
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:23]: This should be Test case #4.

File: docs/PFC-test-plan.md [hunk 9]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
    * Keysight Rx port should receive all the 'Background data traffic' as well as 'Test data traffic'. There should not be any loss observed.
 5. Stop the PFC PAUSE storm.
 6. Repeat the test with a different lossless priority value (!=Pi).
+   
+### Test Case #3 - GLOBAL PAUSE (IEEE 802.3x link level Flow Control)
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:259]: Test case #4

File: docs/PFC-test-plan.md [hunk 10]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+- On SONiC Device Under Test (DUT) configure lossless priority of required values Pi ( e.g., 3 and 4. )
+- Configure following traffic items on the Keysight device:
+  1. Test data traffic: A traffic item from the Keysight Tx port to Rx port. The traffic item uses all the 64 DSCP values (0-63). PFC is enabled at all the 8 priorities of Tx port. The traffic demand is 100% line rate.
+  2. GLOBAL PAUSE storm: Persistent Global pause frames from the Keysight Rx port to the Keysight Tx port. And the inter-frame transmission interval should be smaller than per-frame pause duration.
+   
+#### Test Steps
+
+1. Start GLOBAL PAUSE storm.
+2. After a fixed duration (eg. 1 sec), start the Test data traffic.
+3. Keep the Test data traffic running for a fixed duration (eg 5sec) and then stop the traffic item.
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:276]: 5sec -> 5 sec

File: docs/PFC-test-plan.md [hunk 11]
```diff
@@ -251,3 +255,26 @@ Verify Device Under Test (DUT) processes the PFC PAUSE frame with lossy prioriti
...
+4. Stop the GLOBAL PAUSE storm.
+5. Verify the following:
+   * Keysight Rx port should receive all the 'Test data traffic'. There should not be any loss observed. Throughput should be close to link capacity.
+   * SONiC Device Under Test (DUT) will discard all Global pause frames. However presently there is no way to check if SONiC will discard all global pause frames.
```
Comment (baiwei0427) [inline docs/PFC-test-plan.md:280]: Remove the second item.

**Review by baiwei0427 — CHANGES_REQUESTED** [review]

File: docs/PFC-test-plan.md (hunk 8 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: done

File: docs/PFC-test-plan.md (hunk 10 above)
Comment (SubhajitPalKeysight) [reply docs/PFC-test-plan.md]: done

**Review by baiwei0427 — APPROVED** [review]

**Pull request merged**